server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -2842,17 +2843,19 @@ public AtomicLong getScannedCounter() {
   }
 
   private static String createTabletDirectory(ServerContext context, VolumeManager fs,
-      Table.ID tableId, Text endRow) {
+      Table.ID tableId, KeyExtent extent) {
     String lowDirectory;
 
     UniqueNameAllocator namer = context.getUniqueNameAllocator();
-    VolumeChooserEnvironment chooserEnv = new VolumeChooserEnvironment(tableId, context);
+
+    VolumeChooserEnvironment chooserEnv = new VolumeChooserEnvironment(tableId,
+        new TabletIdImpl(extent), context);
     String volume = fs.choose(chooserEnv, ServerConstants.getBaseUris(context.getConfiguration()))
         + Constants.HDFS_TABLES_DIR + Path.SEPARATOR;
 
     while (true) {
       try {
-        if (endRow == null) {
+        if (extent.getEndRow() == null) {
```

Comment:
Not sure why this would change.

Reply:
Same as above. Now that I am passing the extent down, we need to get the endrow as that is what we wanted. I can mover the getEndRow call up and set it to a local variable if you would prefer

Reply:
I understand now. It wasn't immediately obvious that the change from midRow to low was based on `midRow==low.endRow()` because I'm reviewing on mobile and I have a small screen. I was confused by the parameter type change which I couldn't see at the same time. I thought we were just change which row was being passed. What you have is fine.